lib/src/nesting_builder.dart

```diff
@@ -80,20 +80,11 @@ class NestingBuilder {
   /// If omitted, [spaces] defaults to [Indent.block].
   void indent([int spaces]) {
     if (spaces == null) spaces = Indent.block;
-
-    // Indentation should only change outside of nesting.
-    assert(_pendingNesting == null);
-    assert(_nesting.indent == 0);
-
     _stack.add(_stack.last + spaces);
   }
 
   /// Discards the most recent indentation level.
   void unindent() {
-    // Indentation should only change outside of nesting.
-    assert(_pendingNesting == null);
-    assert(_nesting.indent == 0);
```

Comment:
These sanity checks are useful. If they are tripping, then it's likely you're getting the formatter into a state where it will produce weird output in some cases. You may not have seen those cases in the examples you tested, but they're probably lurking out there.

If at all possible, you'll want to change the visitor code so that these don't fire.